### PR TITLE
feat(vagrant): use "tdp-vagrant" repository as git submodule

### LINKED ISSUE #156: Add "vagrant" feature by using "tdp-vagrant" repository
Use https://github.com/TOSIT-IO/tdp-vagrant as git submodule to deploy VM with vagrant if the `-e vagrant` is defined for the `setup` script.

### PR DESCRIPTION
**Pull request by rpignolet**
We should not run `vagrant` command in parallel leading to undefined behavior. For example, launching multiple `vagrant up ...` can create multiple `vboxnet<N>` with the same network.

I didn't test all playbooks and I think playbooks in `tdp-collection-extras` which perform "prerequisites" will fail. To test it I must add the HTTP proxy support.

#### Which issue(s) this PR fixes:

Fixes #156 

#### Additional comments:

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0) and that you give permission to [TOSIT](https://www.tosit.io/),
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by rpignolet** [comment]
Last force push I removed the `remote_user = vagrant` from `ansible.cfg` which is not needed and the remote user is linked to the infrastructure and we want `ansible.cfg` common to all infrastructure.

**Comment by rpignolet** [comment]
Last force push I removed the `vagrant box add centos/7 --provider virtualbox` command inside the `README.md`. `vagrant up` will download the box if needed.

File: README.md [hunk 1]
```diff
@@ -1,41 +1,40 @@
...
+The Ansible `host.ini` file will be generated using the `hosts` variable in `tdp-vagrant/vagrant.yml`.
 
 ```bash
 # Clone project from version control
 git clone https://github.com/TOSIT-IO/tdp-getting-started.git
 # Move into project dir
 cd tdp-getting-started
-# Setup local env with stable tdp-collection and tdp-collection-extras
-./scripts/setup.sh -e extras
+# Setup local env with stable tdp-collection, tdp-collection-extras and vagrant
```
Comment (gboutry) [inline README.md:32]:
```suggestion
# Setup local env with stable tdp-collection, tdp-collection-extras, and vagrant
```

File: README.md [hunk 2]
```diff
@@ -68,21 +67,24 @@ Execute the `setup.sh` script to create the project directories needed and clone
 ```bash
 # Get stable tdp-collection
 ./scripts/setup.sh
-# Get latest tdp-collection and tdp-collection-extras
-./scripts/setup.sh -e extras -r latest
+# Get latest tdp-collection, tdp-collection-extras and vagrant
```
Comment (gboutry) [inline README.md:70]:
```suggestion
# Get latest tdp-collection, tdp-collection-extras, and vagrant
```

Comment (rpignolet) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (rpignolet) [reply README.md]: done

**Review by gboutry — APPROVED** [review]

**Comment by rpignolet** [comment]
Last force push update `tdp-vagrant` submodule stable commit.

File: .gitmodules [hunk 3]
```diff
@@ -4,3 +4,6 @@
 [submodule "ansible_roles/collections/ansible_collections/tosit/tdp_extra"]
 	path = ansible_roles/collections/ansible_collections/tosit/tdp_extra
 	url = https://github.com/TOSIT-IO/tdp-collection-extras
+[submodule "tdp-vagrant"]
+	path = tdp-vagrant
```
Comment (leopaul36) [inline .gitmodules:8]: What was the logic for cloning `tdp-vagrant` here and not in `ansible_roles` ?

Comment (rpignolet) [reply .gitmodules:8]: `tdp-vagrant` is neither a collection nor a role

**Review by leopaul36 — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -68,21 +67,24 @@ Execute the `setup.sh` script to create the project directories needed and clone
...
-```
+#### Use TDP Vagrant
+
+You can define `vagrant.yml` file to update the machine resources according to your machine's RAM and core count (3Gb of RAM and 4 cores is ideal for the master nodes). The file `tdp-vagrant/vagrant.yml` contains default values.
 
-This playbook will generate the `Vagrantfile` and the `inventory/hosts` file.
+```bash
+cp tdp-vagrant/vagrant.yml .
+# Now you can edit ./vagrant.yml
+```
```
[the remark this answers is not in the document]
Comment (rpignolet) [reply README.md:83]: I wrote line 87 to see TDP Vagrant usage inside the `tdp-vagrant` repository documentation where `vagrant` commands are documented. I can add `vagrant up` if you want.

[the remark this answers is not in the document]
Comment (rpignolet) [reply README.md:83]: I added the `vagrant up` command.

[the remark this answers is not in the document]
Comment (rpignolet) [reply README.md:83]: It is ok for you @leopaul36 or I need to change something ?

**Pull request merged**